File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/DepositCreateRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
public class DepositCreateRequest {
    private UUID roomId;
    private String phone;
    private String tenantFullName;
    private BigDecimal depositAmount;
    private LocalDate expectedMoveInDate;
    private String note;
}

File: Flutter_QuanLyTro/lib/data/models/request/deposit_create_request.dart
class DepositCreateRequest {
  final String roomId;
  final String phone;
  final String tenantFullName;
  final double depositAmount;
  final String expectedMoveInDate;
  final String? note;

  DepositCreateRequest({
    required this.roomId,
    required this.phone,
    required this.tenantFullName,
    required this.depositAmount,
    required this.expectedMoveInDate,
    this.note,
  });

  Map<String, dynamic> toJson() {
    return {
      "roomId": roomId,
      "phone": phone,
      "tenantFullName": tenantFullName,
      "depositAmount": depositAmount,
      "expectedMoveInDate": expectedMoveInDate,
      "note": note,
    };
  }
}